backend/controllers/prisonerProfile/prisonerLocationHistory.js

```diff
@@ -12,7 +12,37 @@ const {
   app: { notmEndpointUrl: dpsUrl },
 } = require('../../config')
 
-module.exports = ({ prisonApi, logError }) => async (req, res) => {
+const fetchStaffName = (context, staffId, prisonApi) => {
+  return prisonApi
+    .getStaffDetails(context, staffId)
+    .then(staff => formatName(staff.firstName, staff.lastName))
+    .catch('No staff name found')
+}
+
+const fetchReasonDescription = (context, assignmentReasonCode, caseNotesApi) => {
+  return caseNotesApi
+    .getCaseNoteTypes(context)
+    .then(caseNoteTypes => caseNoteTypes.find(type => type.code === 'MOVED_CELL'))
+    .then(cellMoveTypes => {
+      return cellMoveTypes?.subCodes.map(subType => ({
+        value: subType.code,
+        text: subType.description,
+      }))
+    })
+    .then(cellMoveReasonRadioValues => cellMoveReasonRadioValues.find(record => record.value === assignmentReasonCode))
```

Comment:
Could the result of the then on line 26 result in an undefined value? This could mean that cellMoveReasonRadioValues is undefined.